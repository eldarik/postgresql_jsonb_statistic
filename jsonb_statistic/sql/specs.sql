CREATE EXTENSION jsonb_statistic;

CREATE TABLE jsonb_test(x jsonb);

INSERT INTO jsonb_test VALUES
('{
  "key": "value",
  "key_1": {
    "key_2": {
      "key_3": "value"
    }
  }
}'),
('{
  "key": "value",
  "key_1": {
    "key_2": "value"
  }
}');

SELECT generate_statistic('{{"jsonb_test", "x"}}');

SELECT * from jsonb_statistics;

DROP TABLE jsonb_test;

DROP EXTENSION jsonb_statistic;
