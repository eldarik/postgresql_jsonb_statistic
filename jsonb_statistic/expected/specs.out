CREATE EXTENSION jsonb_statistic;
CREATE TABLE jsonb_test(x jsonb);
INSERT INTO jsonb_test VALUES
('{
  "key": "value",
  "key_1": {
    "key_2": {
      "key_4": false,
      "key_5": true,
      "key_6": 4.5,
      "key_7": 10000
    }
  }
}'),
('{
  "key": "value",
  "key_1": {
    "key_2": "value"
  }
}');
SELECT generate_statistic('{{"jsonb_test", "x"}}');
 generate_statistic 
--------------------
 
(1 row)

SELECT * from jsonb_statistics;
 tablename  | attname |       key_path        |  value  | value_type | values_count 
------------+---------+-----------------------+---------+------------+--------------
 jsonb_test | x       | ->key_1->key_2->key_4 | false   | boolean    |            1
 jsonb_test | x       | ->key_1->key_2->key_5 | true    | boolean    |            1
 jsonb_test | x       | ->key_1->key_2->key_6 | 4.5     | number     |            1
 jsonb_test | x       | ->key_1->key_2->key_7 | 10000   | number     |            1
 jsonb_test | x       | ->key                 | "value" | string     |            2
 jsonb_test | x       | ->key_1->key_2        | "value" | string     |            1
(6 rows)

DROP TABLE jsonb_test;
DROP EXTENSION jsonb_statistic;
